hibernate-core/src/main/java/org/hibernate/sql/results/graph/collection/internal/AbstractImmediateCollectionInitializer.java

```diff
@@ -71,6 +71,9 @@ public void resolveInstance(RowProcessingState rowProcessingState) {
 		if ( collectionInstance != null || collectionKey == null) {
 			return;
 		}
+		if ( parentAccess != null && !rowProcessingState.isQueryExecution() && parentAccess.findFirstEntityInitializer().isInitialized() ) {
```

Comment:
What happens when the entity in the persistence context has an uninitialized collection? Aren't we "wrongly" skipping initialization then?